serde_arrow/src/test_impls/chrono.rs

```diff
@@ -368,12 +378,76 @@ fn utc_str_as_date64_as_timestamp() {
         .serialize(&items)
         .deserialize(&items)
         .check_nulls(&[&[false, false]]);
+
+    Test::new()
+        .with_schema(json!([{
+            "name": "item",
+            "data_type": "Timestamp(Microsecond, Some(\"Utc\"))",
+            "strategy": "UtcStrAsDate64",
+        }]))
+        .serialize(&items)
+        .deserialize(&items)
+        .check_nulls(&[&[false, false]]);
+
+    Test::new()
+        .with_schema(json!([{
+            "name": "item",
+            "data_type": "Timestamp(Nanosecond, Some(\"Utc\"))",
+            "strategy": "UtcStrAsDate64",
+        }]))
+        .serialize(&items)
+        .deserialize(&items)
+        .check_nulls(&[&[false, false]]);
+
 }
 
 #[test]
 fn naive_as_timestamp() {
     #[allow(deprecated)]
     let items = [
+        Item(NaiveDateTime::from_timestamp_millis(1662921288001).unwrap()),
+        Item(NaiveDateTime::from_timestamp_millis(-2208936075001).unwrap()),
+    ];
+
+    Test::new()
+        .with_schema(json!([{
+            "name": "item",
+            "data_type":
+            "Timestamp(Millisecond, None)",
+            "strategy": "NaiveStrAsDate64",
+        }]))
+        .serialize(&items)
+        .deserialize(&items)
+        .check_nulls(&[&[false, false]]);
+
+    Test::new()
+        .with_schema(json!([{
+            "name": "item",
+            "data_type":
+            "Timestamp(Microsecond, None)",
+            "strategy": "NaiveStrAsDate64",
+        }]))
+        .serialize(&items)
+        .deserialize(&items)
+        .check_nulls(&[&[false, false]]);
+
+    Test::new()
+        .with_schema(json!([{
+            "name": "item",
+            "data_type":
+            "Timestamp(Nanosecond, None)",
+            "strategy": "NaiveStrAsDate64",
+        }]))
+        .serialize(&items)
+        .deserialize(&items)
+        .check_nulls(&[&[false, false]]);
+
+}
+
+#[test]
+fn naive_as_timestamp_seconds() {
+    #[allow(deprecated)]
+        let items = [
         Item(NaiveDateTime::from_timestamp_millis(1662921288000).unwrap()),
```

Comment:
I don't exactly get the +/-1 diff in the timetsamps and why second is different from the other units. 

Could you add a comment to explain this?

Reply:
Done. I wanted to test timestamps with fractional seconds, since this is the part where I found bugs, so I had to special-case timestamps encoded as seconds, since this representation doesn't support fractional seconds.